base/src/commonMain/kotlin/utils/ByteUtils.kt

```diff
@@ -22,4 +22,14 @@ package com.splendo.kaluga.base.utils
  * @param separator The separator to use between elements
  * @return The String representing the [ByteArray]s hexadecimal value
  */
-fun ByteArray.toHexString(separator: CharSequence = "") = asUByteArray().joinToString(separator) { it.toString(16).padStart(2, '0') }
+fun ByteArray.toHexString(separator: CharSequence = "") = asUByteArray()
+    .joinToString(separator) { it.toString(16).padStart(2, '0') }
+
+/**
+ * Converts a hex [String] to [ByteArray]
+ * @return The [ByteArray] or `null` if length is not even
+ */
+fun String.decodeHex(): ByteArray? {
+    if (length % 2 != 0) return null
+    return chunked(2) { it.toString().toInt(16).toByte() }.toByteArray()
```

Comment:
Not important, but you could move 2 into const `BYTE_SPLIT` (or something like that) and  16 into `HEX_BASE`.